### PR TITLE
Personal Footprint

### PR DESCRIPTION
**Pull request by tkamin**
To do:

- [x] Timezones, why is it so hard in Pandas to just work in UTC??? 
- [x] Charting
- [x] Abstration Violations
- [x] Other Cleanup

Kind of working:

Screenshot 2024-08-20 at 6 10 41 PM
Screenshot 2024-08-20 at 6 10 37 PM

### CODE REVIEW

File: power_dashboard/app.py [hunk 1]
```diff
@@ -22,7 +25,7 @@
     page_title="Clean Electricity Dashboard", layout="wide", page_icon=":thunderbolt:"
 )
 
-tab1, tab2 = st.tabs(["Now", "Forecast"])
+tab1, tab2, tab3 = st.tabs(["Now", "Forecast", "Personal"])
```
Comment (ahasha) [inline power_dashboard/app.py:28]: Maybe, "Personal Footprint"?

File: power_dashboard/app.py [hunk 2]
```diff
@@ -339,3 +342,38 @@ def load_forecast_model():
         ax.set_ylabel("Carbon Intensity (gCO2eq/kWh)")
         fig.autofmt_xdate()
         st.pyplot(fig)
+
+    with tab3:
+        st.title("Personal CO2 Calculator from Green Button XML File")
+
+        uploaded_file = st.file_uploader("Choose a Green Button XML file")
```
Comment (ahasha) [inline power_dashboard/app.py:349]: I tested the uploader on my machine and it successfully displayed the table.  I'm taking this is still in a draft state and you're working on combining it with the CO2 intensity history to produce a footprint?

Comment (tkamin) [reply power_dashboard/app.py:349]: Oh, great! Thank you. Yes, this is a draft. I am working on the CO history now

File: power_dashboard/app.py [hunk 3]
```diff
@@ -11,18 +11,23 @@
...
+
+from greenbutton import parse
 
 from power_dashboard.electricity_maps import (
     get_electricity_maps_carbon_intensity,
     get_electricity_maps_power_breakdown,
     get_electricity_maps_zones,
 )
 
+from power_dashboard.eia_api import *
```
Comment (ahasha) [inline power_dashboard/app.py:24]: Better to be explicit about what's imported from this module.

File: power_dashboard/eia_api.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+            "CO2/(kWh)"
+        ].sum()
+    ).reset_index()
+    #st.text(co2_kwh_est_sum.dtypes)
+    co2_kwh_est_sum["timestamp"] = co2_kwh_est_sum["timestamp"].apply(
+        pd.to_datetime, format="%Y/%m/%d %H:%M:%S", utc=True
+    )
+    #st.text(co2_kwh_est_sum.dtypes)
+
+    return co2_kwh_est_sum
```
Comment (ahasha) [inline power_dashboard/eia_api.py:130]: The gridemissions API is only serving data before 6/15 in the bulk files and from 7/20 and 7/21 via API. However, I was able to run his pipeline to generate bulk files covering 6/15 to present. Ideally, I would load this to the gridemissions-ts table and we could pull it from there, so the same co2 intensity algorithm is used consistently throughout. Not a blocker for wrapping up our submission, though!

**Review by ahasha — APPROVED** [review]
LGTM!

**Pull request merged**